lib/vector-tap/src/lib.rs

```diff
@@ -171,57 +183,194 @@ impl<'a> TapRunner<'a> {
-                                            ev.component_id.as_ref(),
-                                            ev.component_kind.as_ref(),
-                                            ev.component_type.as_ref(),
-                                            ev.string.as_ref()
-                                        )
-                                    );
-                                }
-                                GraphQLTapOutputEvent::Trace(ev) => {
-                                    println!(
-                                        "{}",
-                                        self.formatter.format(
-                                            ev.component_id.as_ref(),
-                                            ev.component_kind.as_ref(),
-                                            ev.component_type.as_ref(),
-                                            ev.string.as_ref()
-                                        )
-                                    );
-                                }
-                                GraphQLTapOutputEvent::EventNotification(ev) => {
-                                    if !quiet {
-                                        eprintln!("{}", ev.message);
-                                    }
+                        let output_events: Vec<GraphQLTapOutputEvent> = d
+                            .output_events_by_component_id_patterns
+                            .into_iter()
+                            .filter(|event| {
+                                !matches!(
+                                    (quiet, event),
+                                    (true, GraphQLTapOutputEvent::EventNotification(_))
+                                )
+                            })
+                            .collect();
+
+                        match &self.output_channel {
+                            OutputChannel::Stdout(formatter) => {
+                                self.output_event_stdout(&output_events, formatter);
+                            }
+                            OutputChannel::AsyncChannel(sender_tx) => {
+                                if sender_tx.send(output_events).await.is_err() {
+                                    debug!("Could not send events");
                                 }
                             }
                         }
                     }
                 }
-                Err(_) => {
-                    // If the stream times out, that indicates the duration specified by the user
-                    // has elapsed. We should exit gracefully.
-                    return Ok(());
+                Err(_) =>
+                // If the stream times out, that indicates the duration specified by the user
+                // has elapsed. We should exit gracefully.
+                {
+                    return Ok(())
                 }
                 Ok(_) => return Err(TapExecutorError::GraphQLError),
```

Comment:
💬 A potential improvement here is to change this [fn response](https://github.com/vectordotdev/vector/blob/e9b4fe15b5a84ee9cd2459a70ca92e6c634d519a/lib/vector-api-client/src/subscription.rs#L72-L80) to return the actual error. It doesn't have to be included in this PR.

Reply:
Agreed, will do this in a follow-up PR